ext/ddtrace.c

```diff
@@ -867,6 +873,13 @@ static PHP_MINIT_FUNCTION(ddtrace) {
 
     ddtrace_integrations_minit();
     dd_ip_extraction_startup();
+    ddtrace_user_request_startup();
+
+    // we need to trigger the slow path in the fcall handler in order to be able to suppress calls
+    // otherwise OPLINE is not refreshed after calling the begin observers
+    if (zend_execute_ex == execute_ex) {
+        zend_execute_ex = ddtrace_execute_ex;
+    }
```

Comment:
There's another thing we could leverage though, and it's setting `EG(vm_interrupt) = true;` in the start handler.
This will ensure `zend_interrupt_helper` is called, which will do a `ZEND_VM_ENTER()` reloading the opline. Note that `zend_interrupt_function` must be non-NULL for that, but that's not a problem - just set it to a dummy empty function if it's not NULL yet in post_startup_cb.

Reply:
Also note: needs to be PHP 8.0+ only.

Reply:
it's a bit more complicated because of a SAVE_OPLINE before calling the interrupt function and the fact that the interrupt function is global, but I think it should be right now.

Reply:
You're right, I missed that. But looks good to me now :-)